Linux/jenkins-node/docker/jenkins_agent

```diff
@@ -29,64 +29,89 @@
 # * JENKINS_SECRET : agent secret, if not set as an argument
 # * JENKINS_AGENT_NAME : agent name, if not set as an argument
 # * JENKINS_AGENT_WORKDIR : agent work directory, if not set by optional parameter -workDir
+# * JENKINS_DIRECT_CONNECTION: Connect directly to this TCP agent port, skipping the HTTP(S) connection parameter download.
+#                              Value: "<HOST>:<PORT>"
+# * JENKINS_INSTANCE_IDENTITY: The base64 encoded InstanceIdentity byte array of the Jenkins master. When this is set,
+#                              the agent skips connecting to an HTTP(S) port for connection info.
+# * JENKINS_PROTOCOLS:         Specify the remoting protocols to attempt when instanceIdentity is provided.
 
-# if -tunnel is not provided, try env vars
-case "$@" in
-  *"-tunnel "*) ;;
-  *)
-  if [ ! -z "$JENKINS_TUNNEL" ]; then
-    TUNNEL="-tunnel $JENKINS_TUNNEL"
-  fi ;;
-esac
-
-# if -workDir is not provided, try env vars
-if [ ! -z "$JENKINS_AGENT_WORKDIR" ]; then
-  case "$@" in
-    *"-workDir"*) echo "Warning: Work directory is defined twice in command-line arguments and the environment variable" ;;
-    *)
-    WORKDIR="-workDir $JENKINS_AGENT_WORKDIR" ;;
-  esac
-fi
-
-if [ -n "$JENKINS_URL" ]; then
-  URL="-url $JENKINS_URL"
-fi
-
-if [ -n "$JENKINS_NAME" ]; then
-  JENKINS_AGENT_NAME="$JENKINS_NAME"
-fi  
-
-if [ -z "$JNLP_PROTOCOL_OPTS" ]; then
-  echo "Warning: JnlpProtocol3 is disabled by default, use JNLP_PROTOCOL_OPTS to alter the behavior"
-  JNLP_PROTOCOL_OPTS="-Dorg.jenkinsci.remoting.engine.JnlpProtocol3.disabled=true"
-fi
-
-# if java home is defined, use it
-JAVA_BIN="java"
-if [ "$JAVA_HOME" ]; then
-  JAVA_BIN="$JAVA_HOME/bin/java"
-fi
-
-# if both required options are defined, do not pass the parameters
-OPT_JENKINS_SECRET=""
-if [ -n "$JENKINS_SECRET" ]; then
-  case "$@" in
-    *"${JENKINS_SECRET}"*) echo "Warning: SECRET is defined twice in command-line arguments and the environment variable" ;;
-    *)
-    OPT_JENKINS_SECRET="${JENKINS_SECRET}" ;;
-  esac
-fi
-
-OPT_JENKINS_AGENT_NAME=""
-if [ -n "$JENKINS_AGENT_NAME" ]; then
-  case "$@" in
-    *"${JENKINS_AGENT_NAME}"*) echo "Warning: AGENT_NAME is defined twice in command-line arguments and the environment variable" ;;
-    *)
-    OPT_JENKINS_AGENT_NAME="${JENKINS_AGENT_NAME}" ;;
-  esac
-fi
-
-#TODO: Handle the case when the command-line and Environment variable contain different values.
-#It is fine it blows up for now since it should lead to an error anyway.
-
-exec $JAVA_BIN $JAVA_OPTS $JNLP_PROTOCOL_OPTS -cp /usr/share/jenkins/agent.jar hudson.remoting.jnlp.Main -headless $TUNNEL $URL $WORKDIR $OPT_JENKINS_SECRET $OPT_JENKINS_AGENT_NAME "$@"
+if [ $# -eq 1 ]; then
+
+	# if `docker run` only has one arguments, we assume user is running alternate command like `bash` to inspect the image
```

Comment:
The indentation in this file seems a bit odd - it must be using two tabs instead of two spaces. Can you change this to use two spaces?

Reply:
This is just a direct copy of the `jenkins_agent` that CloudBees provides, so I don't think it's worth touching this.